bus-mapping/src/circuit_input_builder/input_state_ref.rs

```diff
@@ -789,9 +794,11 @@ impl<'a> CircuitInputStateRef<'a> {
                     false,
                     op,
                 );
-                self.tx.steps_mut()[step_index]
-                    .bus_mapping_instance
-                    .push(rev_op_ref);
+                if self.tx.steps().len() > step_index {
```

Comment:
good question:  it is because in the buss mapping use `state.push_op_reversible` for tx access list account warm checking..  in this method if current call is failed, it will add reversion group of current `Op` (refer to https://github.com/privacy-scaling-explorations/zkevm-circuits/blob/main/bus-mapping/src/circuit_input_builder/input_state_ref.rs#L185) , for example, for oog call cases, geth steps in tx can be "`push 100 push 0xfefe ... push Call(OOG_Call step) " ` the last step will add into reversion of key & value pair `(tx.steps.len(), Op)`.  hence when loop reversion group, `self.tx.steps()[tx.steps.len()]` will corrupt . 
I have tried if not use `push_op_reversible` and take use of read operations without push to reversion group, don't need this `if` judgement any more. 
Both seems OK to me , what do you think ? @adria0

Reply:
I will ping @ed255 as CC is on holiday this week, maybe not use `state.push_op_reversible` but only read is better to avoid this `if` statement.

Reply:
sure, will update it.

Reply:
updated to read  in [7ca4a2f](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/827/commits/7ca4a2fa8ddc5f740be472d36309b5323495f0e9)